parallel-install/pkg/config/kubeconfigmanager.go

```diff
@@ -0,0 +1,107 @@
+package config
+
+import (
+	"io/ioutil"
+	"os"
+
+	"github.com/pkg/errors"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+const (
+	temporaryFilePattern = "kubeconfig-*.yaml"
+)
+
+// kubeConfigManager manages resolving kubeconfig from two possible sources: a file path or a kubeconfig content.
+type kubeConfigManager struct {
+	kubeconfigSource KubeconfigSource
+	temporaryPath    string // Mutable!
```

Comment:
This field can be deleted and replaced with a variable in the only function it is used.
This would also automatically make it thread safe, which is not the case at the moment.

Reply:
We create a separate `KubeconfigManager` for every client that needs kubeconfig at the moment. For a helm use case - it is even a matter of a method. It means, when you want to deploy a release using our helm client then it will always create a new manager with a single temporary file and remove it after the deploy is done. The same to release uninstall. We do not change the path in any way when it is passed to the method that uses concurrency under the hood.

Reply:
Still, why have an unnecessary field in the type?